From 1fcd57eb5285b57317dc1c382e813a00d41dce3e Mon Sep 17 00:00:00 2001
From: Rafael Palomar <rafael.palomar@rr-research.no>
Date: Mon, 9 Mar 2020 09:24:23 +0100
Subject: [PATCH] ENH: Improve directories configuration

These improvements try to conciliate configurations between install and the built trees
---
 CMake/SlicerDirectories.cmake      |  1 -
 CMake/SlicerInstallConfig.cmake.in | 10 ++++++++++
 CMake/UseSlicer.cmake.in           | 16 ----------------
 CMake/vtkSlicerConfigure.h.in      | 26 +++++++++++++-------------
 4 files changed, 23 insertions(+), 30 deletions(-)

diff --git a/CMake/SlicerDirectories.cmake b/CMake/SlicerDirectories.cmake
index 657b5fa2a..1ffabc210 100644
--- a/CMake/SlicerDirectories.cmake
+++ b/CMake/SlicerDirectories.cmake
@@ -180,7 +180,6 @@ if(NOT DEFINED Slicer_INSTALL_QTLOADABLEMODULES_SHARE_DIR)
   set(Slicer_INSTALL_QTLOADABLEMODULES_SHARE_DIR "${Slicer_INSTALL_ROOT}${Slicer_QTLOADABLEMODULES_SHARE_DIR}")
 endif()
 
-
 #-----------------------------------------------------------------------------
 # Scripted Modules relative directories
 #-----------------------------------------------------------------------------
diff --git a/CMake/SlicerInstallConfig.cmake.in b/CMake/SlicerInstallConfig.cmake.in
index 1f69327d8..174ef7664 100644
--- a/CMake/SlicerInstallConfig.cmake.in
+++ b/CMake/SlicerInstallConfig.cmake.in
@@ -143,6 +143,16 @@ set(Slicer_INSTALL_QTLOADABLEMODULES_LIB_DIR "${Slicer_HOME}/@Slicer_INSTALL_QTL
 set(Slicer_INSTALL_QTLOADABLEMODULES_INCLUDE_DIR "${Slicer_HOME}/@Slicer_INSTALL_QTLOADABLEMODULES_INCLUDE_DIR@")
 set(Slicer_INSTALL_QTLOADABLEMODULES_SHARE_DIR "${Slicer_HOME}/@Slicer_INSTALL_QTLOADABLEMODULES_SHARE_DIR@")
 
+set(Slicer_QTSCRIPTEDMODULES_SUBDIR "@Slicer_QTSCRIPTEDMODULES_SUBDIR@")
+set(Slicer_QTSCRIPTEDMODULES_BIN_DIR "@Slicer_QTSCRIPTEDMODULES_BIN_DIR@")
+set(Slicer_QTSCRIPTEDMODULES_LIB_DIR "@Slicer_QTSCRIPTEDMODULES_LIB_DIR@")
+set(Slicer_QTSCRIPTEDMODULES_INCLUDE_DIR "@Slicer_QTSCRIPTEDMODULES_INCLUDE_DIR@")
+set(Slicer_QTSCRIPTEDMODULES_SHARE_DIR "@Slicer_QTSCRIPTEDMODULES_SHARE_DIR@")
+
+set(Slicer_INSTALL_QTSCRIPTEDMODULES_BIN_DIR "${Slicer_HOME}/@Slicer_INSTALL_QTSCRIPTEDMODULES_BIN_DIR@")
+set(Slicer_INSTALL_QTSCRIPTEDMODULES_LIB_DIR "${Slicer_HOME}/@Slicer_INSTALL_QTSCRIPTEDMODULES_LIB_DIR@")
+set(Slicer_INSTALL_QTSCRIPTEDMODULES_INCLUDE_DIR "${Slicer_HOME}/@Slicer_INSTALL_QTSCRIPTEDMODULES_INCLUDE_DIR@")
+set(Slicer_INSTALL_QTSCRIPTEDMODULES_SHARE_DIR "${Slicer_HOME}/@Slicer_INSTALL_QTSCRIPTEDMODULES_SHARE_DIR@")
 
 # Components needed to build modules from outside a Slicer build tree or
 # from outside a Slicer installation tree
diff --git a/CMake/UseSlicer.cmake.in b/CMake/UseSlicer.cmake.in
index 241c2d39f..76d79f2ac 100644
--- a/CMake/UseSlicer.cmake.in
+++ b/CMake/UseSlicer.cmake.in
@@ -317,9 +317,6 @@ endif()
 # Set Slicer_INSTALL_*MODULE_*_DIR variables
 #-----------------------------------------------------------------------------
 if(Slicer_BUILD_CLI_SUPPORT)
-  set(Slicer_INSTALL_CLIMODULES_BIN_DIR "${Slicer_HOME}/${Slicer_BUNDLE_EXTENSIONS_LOCATION}${Slicer_CLIMODULES_BIN_DIR}")
-  set(Slicer_INSTALL_CLIMODULES_LIB_DIR "${Slicer_HOME}/${Slicer_BUNDLE_EXTENSIONS_LOCATION}${Slicer_CLIMODULES_LIB_DIR}")
-  set(Slicer_INSTALL_CLIMODULES_SHARE_DIR "${Slicer_HOME}/${Slicer_BUNDLE_EXTENSIONS_LOCATION}${Slicer_CLIMODULES_SHARE_DIR}")
   if(APPLE)
     set(SlicerExecutionModel_DEFAULT_CLI_INSTALL_RUNTIME_DESTINATION "${Slicer_INSTALL_CLIMODULES_BIN_DIR}")
     set(SlicerExecutionModel_DEFAULT_CLI_INSTALL_LIBRARY_DESTINATION "${Slicer_INSTALL_CLIMODULES_LIB_DIR}")
@@ -331,19 +328,6 @@ if(Slicer_BUILD_CLI_SUPPORT)
   endif()
 endif()
 
-set(Slicer_INSTALL_QTLOADABLEMODULES_BIN_DIR "${Slicer_HOME}${Slicer_BUNDLE_EXTENSIONS_LOCATION}/${Slicer_QTLOADABLEMODULES_BIN_DIR}")
-set(Slicer_INSTALL_QTLOADABLEMODULES_LIB_DIR "${Slicer_HOME}${Slicer_BUNDLE_EXTENSIONS_LOCATION}/${Slicer_QTLOADABLEMODULES_LIB_DIR}")
-set(Slicer_INSTALL_QTLOADABLEMODULES_PYTHON_LIB_DIR "${Slicer_HOME}${Slicer_BUNDLE_EXTENSIONS_LOCATION}/${Slicer_QTLOADABLEMODULES_PYTHON_LIB_DIR}")
-set(Slicer_INSTALL_QTLOADABLEMODULES_INCLUDE_DIR "${Slicer_HOME}${Slicer_BUNDLE_EXTENSIONS_LOCATION}/${Slicer_QTLOADABLEMODULES_INCLUDE_DIR}")
-set(Slicer_INSTALL_QTLOADABLEMODULES_SHARE_DIR "${Slicer_HOME}${Slicer_BUNDLE_EXTENSIONS_LOCATION}/${Slicer_QTLOADABLEMODULES_SHARE_DIR}")
-
-if(Slicer_USE_PYTHONQT)
-  set(Slicer_INSTALL_QTSCRIPTEDMODULES_BIN_DIR "${Slicer_HOME}${Slicer_BUNDLE_EXTENSIONS_LOCATION}${Slicer_QTSCRIPTEDMODULES_BIN_DIR}")
-  set(Slicer_INSTALL_QTSCRIPTEDMODULES_LIB_DIR "${Slicer_HOME}${Slicer_BUNDLE_EXTENSIONS_LOCATION}${Slicer_QTSCRIPTEDMODULES_LIB_DIR}")
-  set(Slicer_INSTALL_QTSCRIPTEDMODULES_INCLUDE_DIR "${Slicer_HOME}${Slicer_BUNDLE_EXTENSIONS_LOCATION}${Slicer_QTSCRIPTEDMODULES_INCLUDE_DIR}")
-  set(Slicer_INSTALL_QTSCRIPTEDMODULES_SHARE_DIR "${Slicer_HOME}${Slicer_BUNDLE_EXTENSIONS_LOCATION}${Slicer_QTSCRIPTEDMODULES_SHARE_DIR}")
-endif()
-
 # These variables can be used when configuring extension external projects in
 # two different scenarios: (1) bundled extensions and (2) regular extensions.
 #
diff --git a/CMake/vtkSlicerConfigure.h.in b/CMake/vtkSlicerConfigure.h.in
index 52ce5f5b0..80b223191 100644
--- a/CMake/vtkSlicerConfigure.h.in
+++ b/CMake/vtkSlicerConfigure.h.in
@@ -48,9 +48,9 @@
 #cmakedefine Slicer_BUILD_WIN32_CONSOLE
 
 #define Slicer_MAIN_PROJECT_APPLICATION_NAME "@Slicer_MAIN_PROJECT_APPLICATION_NAME@"
-#define Slicer_BIN_DIR "@Slicer_BIN_DIR@"
+#define Slicer_BIN_DIR "@Slicer_INSTALL_BIN_DIR@"
 #define Slicer_ETC_DIR "@Slicer_ETC_DIR@"
-#define Slicer_LIB_DIR "@Slicer_LIB_DIR@"
+#define Slicer_LIB_DIR "@Slicer_INSTALL_LIB_DIR@"
 #define Slicer_INCLUDE_DIR "@Slicer_INCLUDE_DIR@"
 #define Slicer_SHARE_DIR "@Slicer_SHARE_DIR@"
 #define Slicer_ITKFACTORIES_DIR "@Slicer_ITKFACTORIES_DIR@"
@@ -74,8 +74,8 @@
 #cmakedefine Slicer_BUILD_CLI_SUPPORT
 #cmakedefine Slicer_BUILD_CLI
 #define Slicer_CLIMODULES_SUBDIR "@Slicer_CLIMODULES_SUBDIR@"
-#define Slicer_CLIMODULES_BIN_DIR "@Slicer_CLIMODULES_BIN_DIR@"
-#define Slicer_CLIMODULES_LIB_DIR "@Slicer_CLIMODULES_LIB_DIR@"
+#define Slicer_CLIMODULES_BIN_DIR "@Slicer_INSTALL_CLIMODULES_BIN_DIR@"
+#define Slicer_CLIMODULES_LIB_DIR "@Slicer_INSTALL_CLIMODULES_LIB_DIR@"
 #define Slicer_CLIMODULES_SHARE_DIR "@Slicer_CLIMODULES_SHARE_DIR@"
 
 #define Slicer_CLI_PREFER_EXECUTABLE_DEFAULT true
@@ -89,21 +89,21 @@
 
 #cmakedefine Slicer_BUILD_QTLOADABLEMODULES
 #define Slicer_QTLOADABLEMODULES_SUBDIR "@Slicer_QTLOADABLEMODULES_SUBDIR@"
-#define Slicer_QTLOADABLEMODULES_BIN_DIR "@Slicer_QTLOADABLEMODULES_BIN_DIR@"
-#define Slicer_QTLOADABLEMODULES_LIB_DIR "@Slicer_QTLOADABLEMODULES_LIB_DIR@"
-#define Slicer_QTLOADABLEMODULES_INCLUDE_DIR "@Slicer_QTLOADABLEMODULES_INCLUDE_DIR@"
-#define Slicer_QTLOADABLEMODULES_SHARE_DIR "@Slicer_QTLOADABLEMODULES_SHARE_DIR@"
+#define Slicer_QTLOADABLEMODULES_BIN_DIR "@Slicer_INSTALL_QTLOADABLEMODULES_BIN_DIR@"
+#define Slicer_QTLOADABLEMODULES_LIB_DIR "@Slicer_INSTALL_QTLOADABLEMODULES_LIB_DIR@"
+#define Slicer_QTLOADABLEMODULES_INCLUDE_DIR "@Slicer_INSTALL_QTLOADABLEMODULES_INCLUDE_DIR@"
+#define Slicer_QTLOADABLEMODULES_SHARE_DIR "@Slicer_INSTALL_QTLOADABLEMODULES_SHARE_DIR@"
 #ifdef Slicer_USE_PYTHONQT
-#  define Slicer_QTLOADABLEMODULES_PYTHON_LIB_DIR "@Slicer_QTLOADABLEMODULES_PYTHON_LIB_DIR@"
+#  define Slicer_QTLOADABLEMODULES_PYTHON_LIB_DIR "@Slicer_INSTALL_QTLOADABLEMODULES_PYTHON_LIB_DIR@"
 #endif
 
 #cmakedefine Slicer_BUILD_QTSCRIPTEDMODULES
 #ifdef Slicer_USE_PYTHONQT
 #  define Slicer_QTSCRIPTEDMODULES_SUBDIR "@Slicer_QTSCRIPTEDMODULES_SUBDIR@"
-#  define Slicer_QTSCRIPTEDMODULES_BIN_DIR "@Slicer_QTSCRIPTEDMODULES_BIN_DIR@"
-#  define Slicer_QTSCRIPTEDMODULES_LIB_DIR "@Slicer_QTSCRIPTEDMODULES_LIB_DIR@"
-#  define Slicer_QTSCRIPTEDMODULES_INCLUDE_DIR "@Slicer_QTSCRIPTEDMODULES_INCLUDE_DIR@"
-#  define Slicer_QTSCRIPTEDMODULES_SHARE_DIR "@Slicer_QTSCRIPTEDMODULES_SHARE_DIR@"
+#  define Slicer_QTSCRIPTEDMODULES_BIN_DIR "@Slicer_INSTALL_QTSCRIPTEDMODULES_BIN_DIR@"
+#  define Slicer_QTSCRIPTEDMODULES_LIB_DIR "@Slicer_INSTALL_QTSCRIPTEDMODULES_LIB_DIR@"
+#  define Slicer_QTSCRIPTEDMODULES_INCLUDE_DIR "@Slicer_INSTALL_QTSCRIPTEDMODULES_INCLUDE_DIR@"
+#  define Slicer_QTSCRIPTEDMODULES_SHARE_DIR "@Slicer_INSTALL_QTSCRIPTEDMODULES_SHARE_DIR@"
 #endif //Slicer_USE_PYTHONQT
 
 // ThirdParty: Corresponds to superbuild projects built
-- 
2.24.1

